mongodbatlas/fw_data_source_mongodbatlas_projects.go

```diff
@@ -0,0 +1,202 @@
+							Computed: true,
+						},
+						"teams": schema.ListNestedAttribute{
+							Computed: true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"team_id": schema.StringAttribute{
+										Computed: true,
+									},
+									"role_names": schema.ListAttribute{
+										Computed:    true,
+										ElementType: types.StringType,
+									},
+								},
+							},
+						},
+						"limits": schema.SetNestedAttribute{
+							Computed: true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"name": schema.StringAttribute{
+										Computed: true,
+									},
+									"value": schema.Int64Attribute{
+										Computed: true,
+									},
+									"current_usage": schema.Int64Attribute{
+										Computed: true,
+									},
+									"default_limit": schema.Int64Attribute{
+										Computed: true,
+									},
+									"maximum_limit": schema.Int64Attribute{
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *ProjectsDS) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var stateModel tfProjectsDSModel
+	conn := d.client.Atlas
+	connV2 := d.client.AtlasV2
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &stateModel)...)
+	options := &matlas.ListOptions{
+		PageNum:      int(stateModel.PageNum.ValueInt64()),
+		ItemsPerPage: int(stateModel.ItemsPerPage.ValueInt64()),
+	}
+
+	projectsRes, _, err := conn.Projects.GetAllProjects(ctx, options)
+	if err != nil {
+		resp.Diagnostics.AddError("error in monogbatlas_projects data source", fmt.Sprintf("error getting projects information: %s", err.Error()))
+		return
+	}
+
+	err = populateProjectsDataSourceModel(ctx, conn, connV2, &stateModel, projectsRes)
+	if err != nil {
+		resp.Diagnostics.AddError("error in monogbatlas_projects data source", err.Error())
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func populateProjectsDataSourceModel(ctx context.Context, conn *matlas.Client, connV2 *admin.APIClient, stateModel *tfProjectsDSModel, projectsRes *matlas.Projects) error {
+	results := make([]tfProjectDSModel, len(projectsRes.Results))
+
+	for i, project := range projectsRes.Results {
+		atlasTeams, atlasLimits, atlasProjectSettings, err := getProjectPropsFromAPI(ctx, conn, connV2, project.ID)
+		if err != nil {
+			return fmt.Errorf("error while getting project properties for project %s: %v", project.ID, err.Error())
+		}
+		projectModel := toTFProjectDataSourceModel(ctx, project, atlasTeams, atlasProjectSettings, atlasLimits)
+		results[i] = projectModel
+	}
+
+	stateModel.Results = results
+	stateModel.TotalCount = types.Int64Value(int64(projectsRes.TotalCount))
+	stateModel.ID = types.StringValue(fmt.Sprint(projectsRes.TotalCount))
```

Comment:
This is a dummy value pretty much because acceptance tests require every data source to have an 'id' field. TF will get rid of it at some point but for now they recommend to just add it and include a deprecation note to ensure users don't use it https://github.com/hashicorp/terraform-plugin-testing/issues/84#issuecomment-1480006432